oop-customers/Customer.php

```diff
@@ -0,0 +1,117 @@
+<?php
+declare(strict_types=1);
+system('clear');
+
+class Customer
+{
+    /**
+     * Status of the customer, can be basic or premium.
+     *
+     * @var string
+     */
+    public $status = 'basic';
+
+    /**
+     * Age of the customer
+     *
+     * @var integer
+     */
+    public $age;
+
+    protected $name;
+    protected $surname;
+    protected $nip;
+
+    public function __construct(int $age=0)
+    {
+        $this->age = $age;
+    }
+
+    public function getDiscount(): int
+    {
+        return 20;
+    }
+
+    public function setName(string $name):void
+    {
+        $this->name = $name;
+    }
+
+    public function setSurname(string $surname):void
+    {
+        $this->surname = $surname;
+    }
+
+    public function setNIP(string $nip):void
+    {
+        $this->nip = $nip;
+    }
+
+    /**
+     * Function to show name of the customer.
+     * @return string
+     * @throws Error when customer name is not set.
+     */
+    public function getName(): string
+    {
+        if (!(isset($this->name))) {
+            throw new Error("\e[31mThere is no name for the invoice!\e[0m");
+        } else {
+            return $this->name;
```

Comment:
Tutaj ciekawostka: rzucenie nieprzechwyconego wyjątku przerywa flow aplikacji, więc zapisanie returna w else jest nadmiarowe, można napisać prościej:

`if (warunek) {
    throw new Exception(...);
}

return $this->name;`